### PR TITLE
Small refactor

### PR DESCRIPTION
**Pull request by DrunkWinter**

### CODE REVIEW

File: src/components/innertubeClient.js [hunk 1]
```diff
@@ -29,8 +29,8 @@ function getInnertubeEmbedPayload(videoId, playlistId, playlistIndex) {
     const payload = {
         context: {
             client: {
-                clientName: getYtcfgValue('INNERTUBE_CLIENT_NAME').replace('_EMBEDDED_PLAYER', ''),
-                clientVersion: getYtcfgValue('INNERTUBE_CLIENT_VERSION'),
+                // clientName: getYtcfgValue('INNERTUBE_CLIENT_NAME').replace('_EMBEDDED_PLAYER', ''),
```
Comment (DrunkWinter) [inline src/components/innertubeClient.js:32]: We are populating the object with information, but this entire object gets overridden on line 54?

File: src/components/innertubeClient.js [hunk 2]
```diff
@@ -50,9 +50,9 @@ function getInnertubeEmbedPayload(videoId, playlistId, playlistIndex) {
     const innertubeContext = getYtcfgValue('INNERTUBE_CONTEXT');
 
     // Append client info from YT config (ytcfg)
-    if (typeof innertubeContext?.client === "object") {
+    // if (typeof innertubeContext?.client === "object") {
```
[the remark this answers is not in the document]
Comment (zerodytrash) [reply src/components/innertubeClient.js:53]: true, the check can be removed.

**Comment by zerodytrash** [comment]
looks good

**Comment by DrunkWinter** [comment]
> looks good

How about the second issue? We are populating `client` but then overriding it. So what was the point in the first place?

**Comment by zerodytrash** [comment]
Oh my mistake. The idea was to fill the self created client object with additional attributes from the youtube configuration. I think I didn't realized that the Youtube configuration also overwrites the `CLIENT_NAME` and `CLIENT_VERSION ` attribute. Therefore we can assign the youtube client configuration directly to the payload without a merge in line 54.

**Comment by zerodytrash** [comment]
i don't know if `.replace('_EMBEDDED_PLAYER', '')` is needed anywhere. This is not from me lol. I think for the embed player the script does not work in general. Maybe we should check it.

File: src/components/innertubeClient.js (hunk 1 above)
Comment (zerodytrash) [reply src/components/innertubeClient.js:32]: replace it with `client: getYtcfgValue('INNERTUBE_CONTEXT').client` or something like that

**Comment by DrunkWinter** [comment]
It turns out we need `clientScreen: 'embed'` or else the description and related videos wont be included.

**Comment by DrunkWinter** [comment]
By my testing method, embeds did not work. In that case we need to add `@noframes` in `userscript.config.js`, so we don't run anywhere else other than the YouTube domain.

I'll let you test it also just in case before doing that.

**Comment by zerodytrash** [comment]
> It turns out we need clientScreen: 'embed' or else the description and related videos wont be included.

Oh yes, `clientScreen`:`EMBED` is the key for the bypass. I had forgotten that 😩

> I'll let you test it also just in case before doing that.

I will test this tomorrow

**Comment by zerodytrash** [comment]
> I will test this tomorrow

I found a way to unlock embedded videos as well. We can apply the pull request and then I implement the bypass for embed videos.

**Pull request merged**